providers/implementations/digests/blake2_prov.c

```diff
@@ -35,7 +35,75 @@ IMPLEMENT_digest_functions(blake2s256, BLAKE2S_CTX,
                            ossl_blake2s_final)
 
 /* ossl_blake2b512_functions */
-IMPLEMENT_digest_functions(blake2b512, BLAKE2B_CTX,
-                           BLAKE2B_BLOCKBYTES, BLAKE2B_DIGEST_LENGTH, 0,
-                           ossl_blake2b512_init, ossl_blake2b_update,
-                           ossl_blake2b_final)
+
+static OSSL_FUNC_digest_init_fn blake2b512_internal_init;
+static OSSL_FUNC_digest_newctx_fn blake2b512_newctx;
+static OSSL_FUNC_digest_freectx_fn blake2b512_freectx;
+static OSSL_FUNC_digest_dupctx_fn blake2b512_dupctx;
+static OSSL_FUNC_digest_final_fn blake2b512_internal_final;
+
+static int blake2b512_internal_init(void *ctx, const OSSL_PARAM params[])
+{
+    return ossl_prov_is_running() && ossl_blake2b_set_ctx_params(ctx, params)
+        && ossl_blake2b512_init(ctx);
+}
+
+static void *blake2b512_newctx(void *prov_ctx)
+{
+    struct blake2b_md_data_st *ctx =
+        ossl_prov_is_running()? OPENSSL_zalloc(sizeof(*ctx)) : NULL;
+    return ctx;
```

Comment:
We normally put a blank line between declarations and code.

I guess it would be rewritten:
```c
return ossl_prov_is_running() ? OPENSSL_zalloc(sizeof(struct blake2b_md_data_st)) : NULL;
```

But I think it looks better with just a blank line inserted.